diff --git platform/x11/pdfapp.c platform/x11/pdfapp.c
index 006cba1..71cb137 100644
--- platform/x11/pdfapp.c
+++ platform/x11/pdfapp.c
@@ -1239,6 +1239,10 @@ void pdfapp_onkey(pdfapp_t *app, int c, int modifiers)
 	case 'H':
 		pdfapp_autozoom_vertical(app);
 		break;
+	case 'w':
+		pdfapp_autozoom(app);
+		break;
+
 	case 'Z':
 		pdfapp_autozoom(app);
 		break;
@@ -1288,17 +1292,6 @@ void pdfapp_onkey(pdfapp_t *app, int c, int modifiers)
 		app->fullscreen = !app->fullscreen;
 		break;
 
-	case 'w':
-		if (app->fullscreen)
-		{
-			winfullscreen(app, 0);
-			app->fullscreen = 0;
-		}
-		app->shrinkwrap = 1;
-		app->panx = app->pany = 0;
-		pdfapp_showpage(app, 0, 0, 1, 0, 0);
-		break;
-
 	case 'h':
 		app->panx += fz_pixmap_width(app->ctx, app->image) / 10;
 		pdfapp_showpage(app, 0, 0, 1, 0, 0);
diff --git source/fitz/load-jpx.c source/fitz/load-jpx.c
index 7cd3647..d9848b8 100644
--- source/fitz/load-jpx.c
+++ source/fitz/load-jpx.c
@@ -5,7 +5,7 @@
  * linux toolchains too. */
 #define OPJ_STATIC
 #ifndef _MSC_VER
-#define OPJ_HAVE_STDINT_H
+//#define OPJ_HAVE_STDINT_H
 #endif
 
 #include <openjpeg.h>
@@ -117,7 +117,7 @@ fz_load_jpx(fz_context *ctx, unsigned char *data, int size, fz_colorspace *defcs
 	opj_stream_set_read_function(stream, fz_opj_stream_read);
 	opj_stream_set_skip_function(stream, fz_opj_stream_skip);
 	opj_stream_set_seek_function(stream, fz_opj_stream_seek);
-	opj_stream_set_user_data(stream, &sb);
+	opj_stream_set_user_data(stream, &sb, NULL);
 	/* Set the length to avoid an assert */
 	opj_stream_set_user_data_length(stream, size);
 
